tools/ts/ast_util.ts

```diff
@@ -0,0 +1,237 @@
+import { relative } from "path";
+import { readFileSync } from "fs";
+import Project, {
+  ExportDeclaration,
+  ImportDeclaration,
+  InterfaceDeclaration,
+  JSDoc,
+  NamespaceDeclaration,
+  NamespaceDeclarationKind,
+  PropertySignature,
+  SourceFile,
+  StatementedNode,
+  TypeGuards,
+  VariableStatement,
+  VariableDeclarationKind
+} from "ts-simple-ast";
+
+// It is too complicate to deal with only what is used out of `msg_generated` so
+// we deal with it in a _hardcoded_ way.
+const MSG_GENERATED_SPECIFIER = "gen/msg_generated";
+export const INTERNAL_GENERATED_SPECIFIER = "@internal/generated";
+
+let rootPath = process.cwd();
```

Comment:
Superfluous? Also please avoid top-level functionality - it would be nice if everything was wrapped in functions so we can test it.